go/tools/builders/compilepkg.go

```diff
@@ -449,18 +454,25 @@ func compileArchive(
 		// Add unknown origin source files into the mix.
 		nogoSrcs = append(nogoSrcs, goSrc)
 	}
-	if nogoPath != "" && len(nogoSrcs) > 0 {
-		ctx, cancel := context.WithCancel(context.Background())
-		nogoChan = make(chan error)
-		go func() {
-			nogoChan <- runNogo(ctx, workDir, nogoPath, nogoSrcs, deps, packagePath, importcfgPath, outFactsPath)
-		}()
-		defer func() {
-			if nogoChan != nil {
-				cancel()
-				<-nogoChan
+	if nogoPath != "" {
+		if len(nogoSrcs) == 0 {
```

Comment:
Can we move this check into `runNogo`? It's the function that calls the nogo tool and it should be responsible for making sure that the list of srcs passed to the tool is not empty.

Reply:
That looks much nicer, thanks for the suggestion. In fact, I think that we might be able to replace `runNogo` with a pure Starlark implementation in a follow-up PR.

@tyler-french Could you test this in case @linzhp hasn't yet?